File: problems/67. Add Binary/Solution.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        int lengthDiff = std::abs((int)a.length() - (int)b.length());
        if (a.length() < b.length()) a = string(lengthDiff, '0') + a;
        if (b.length() < a.length()) b = string(lengthDiff, '0') + b;
        int carry = 0;
        string ret = "";
        for (int i = a.length() - 1; i >= 0; i--) {
            int total = (a[i] - '0') + (b[i] - '0') + carry;
            if (total == 3) {
                ret = "1" + ret;
                carry = 1;
            } else if (total == 2) {
                ret = "0" + ret;
                carry = 1;
            } else {
                ret = to_string(total) + ret;
                carry = 0;
            }
        }
        return (carry == 1) ? "1" + ret : ret;
    }
};

File: problems/67. Add Binary/Solution.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        lengthDiff = abs(len(a) - len(b))
        if len(b) < len(a):
            b = lengthDiff * "0" + b
        if len(a) < len(b):
            a = lengthDiff * "0" + a
        ret = ""
        carry = 0
        for i in range(len(b) - 1, -1, -1):
            total = int(a[i]) + int(b[i]) + carry
            if total == 3: # Carry 1
                ret = "1" + ret
                carry = 1
            elif total == 2:
                ret = "0" + ret
                carry = 1
            else:
                ret = str(total) + ret
                carry = 0
        return ret if carry == 0 else "1" + ret